python/lsst/analysis/tools/actions/plot/elements/plotInfoElement.py

```diff
@@ -0,0 +1,146 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("PlotInfoElement",)
+
+from typing import TYPE_CHECKING, Any, Mapping
+
+import matplotlib.transforms as transforms
+
+from ....interfaces import PlotElement
+
+if TYPE_CHECKING:
+    from matplotlib.axes import Axes
+
+
+class PlotInfoElement(PlotElement):
+    """Adds plotInfo to a figure.
+
+    Notes
+    ------
+    Converts a PlotElement dict to a string and creates
+    a textbox to display it.
+    """
+
+    def _buildPlotInfoText(self, plotInfo: Mapping[str, str]) -> str:
+        """Add useful information to the plot.
+
+        Parameters
+        ----------
+            A dictionary of the plot information.
+
+        Returns
+        -------
+        fig : `matplotlib.figure.Figure`
+            The figure with the information added.
+        """
+        # TO DO: This function is a near copy-paste of addPlotInfo
+        #        from plotUtils, but which only builds the plot info
+        #        text, rather than adding it to a figure. Suggest the
+        #        text-build is separated-out from addPlotInfo in plotUtils.
+        # TO DO: figure out how to get this information
+        photocalibDataset = "None"
+        astroDataset = "None"
+
+        run = plotInfo["run"]
+        datasetsUsed = f"\nPhotoCalib: {photocalibDataset}, Astrometry: {astroDataset}"
+        tableType = f"\nTable: {plotInfo['tableName']}"
+
+        dataIdText = ""
+        if "tract" in plotInfo.keys():
+            dataIdText += f", Tract: {plotInfo['tract']}"
+        if "visit" in plotInfo.keys():
+            dataIdText += f", Visit: {plotInfo['visit']}"
+
+        bandText = ""
+        for band in plotInfo["bands"]:
+            bandText += band + ", "
+        bandsText = f", Bands: {bandText[:-2]}"
+        infoText = f"\n{run}{datasetsUsed}{tableType}{dataIdText}{bandsText}"
+
+        # Find S/N and mag keys, if present.
+        snKeys = []
+        magKeys = []
+        selectionKeys = []
+        selectionPrefix = "Selection: "
+        for key, value in plotInfo.items():
+            if "SN" in key or "S/N" in key:
+                snKeys.append(key)
+            elif "Mag" in key:
+                magKeys.append(key)
+            elif key.startswith(selectionPrefix):
+                selectionKeys.append(key)
+        # Add S/N and mag values to label, if present.
+        # TODO: Do something if there are multiple sn/mag keys. Log? Warn?
```

Comment:
I realize this is copied over from the existing code, so I'm not expecting you to address it here necessarily. However, what I would do were it me is to add an optional `logger` argument into this function which is a logging instance. Then, inside this function where appropriate, add `if logger` blocks which then call `logger.info` or `logger.warning` etc when one of these triggers is `True`.